src/main/java/com/marginallyclever/makelangelo/plotter/plotterControls/PlotterControls.java

```diff
@@ -1,90 +1,151 @@
 import com.marginallyclever.makelangelo.turtle.TurtleMove;
 import com.marginallyclever.util.PreferencesHelper;
 
 import javax.swing.*;
+import javax.swing.border.Border;
 import java.awt.*;
 import java.util.List;
 
 public class PlotterControls extends JPanel {
-	private static final long serialVersionUID = 1L;
-	private Plotter myPlotter;
-	private Turtle myTurtle;
-	private JogInterface jogInterface;
-	private MarlinInterface marlinInterface;
-	private ProgramInterface programInterface;
-
-	//private JButton bSaveGCode = new JButton(Translator.get("PlotterControls.SaveGCode"));
-	private JButton bFindHome = new JButton(Translator.get("JogInterface.FindHome"));
-	private JButton bRewind = new JButton(Translator.get("PlotterControls.Rewind"));
-	private JButton bStart = new JButton(Translator.get("PlotterControls.Play"));
-	private JButton bStep = new JButton(Translator.get("PlotterControls.Step"));
-	private JButton bPause = new JButton(Translator.get("PlotterControls.Pause"));
-	private JProgressBar progress = new JProgressBar(0, 100);
+	private final Plotter myPlotter;
+	private final Turtle myTurtle;
+	private final JogInterface jogInterface;
+	private final MarlinInterface marlinInterface;
+	private final ProgramInterface programInterface;
+
+	private JComboBox<NetworkSessionItem> connectionComboBox = new JComboBox<>();
+	private ConnectionButton connectionButton = new ConnectionButton(connectionComboBox);
+	private ButtonIcon bFindHome;
+	private ButtonIcon bRewind;
+	private ButtonIcon bStart;
+	private ButtonIcon bStep;
+	private ButtonIcon bPause;
+	private ButtonIcon bEmergencyStop;
+	private final JProgressBar progress = new JProgressBar(0, 100);
 
 	private boolean isRunning = false;
 	private boolean penIsUpBeforePause = false;
 
-	public PlotterControls(Plotter plotter, Turtle turtle) {
+	public PlotterControls(Plotter plotter, Turtle turtle, Window parentWindow) {
 		super();
 		myPlotter = plotter;
 		myTurtle = turtle;
 
 		jogInterface = new JogInterface(plotter);
-		marlinInterface = new MarlinPlotterInterface(plotter);
+		marlinInterface = new MarlinPlotterInterface(plotter, connectionButton);
 		programInterface = new ProgramInterface(plotter, turtle);
 
-		JTabbedPane pane = new JTabbedPane();
-		pane.addTab(JogInterface.class.getSimpleName(), jogInterface);
-		pane.addTab(MarlinInterface.class.getSimpleName(), marlinInterface);
-		pane.addTab(ProgramInterface.class.getSimpleName(), programInterface);
+		JTabbedPane tabbedPane = new JTabbedPane();
+		jogInterface.setPreferredSize(new Dimension(580, 300));
+		tabbedPane.addTab(Translator.get("PlotterControls.JogTab"), jogInterface);
+		tabbedPane.addTab(Translator.get("PlotterControls.MarlinTab"), marlinInterface);
+		tabbedPane.addTab(Translator.get("PlotterControls.ProgramTab"), programInterface);
+
+		CollapsiblePanel collapsiblePanel = new CollapsiblePanel(parentWindow, Translator.get("PlotterControls.AdvancedControls"), 570);
```

Comment:
parentWindow could be PlotterControls?

Reply:
no, the `CollapsiblePanel` need a dialog box to be able to resize it when opening of collapsing it